salt/elife-xpub/scripts/setup-database.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -e
 
-if ! docker-compose -f docker-compose.yml -f docker-compose.formula.yml exec -T postgres /bin/bash -c "until echo > /dev/tcp/postgres/5432; do sleep 1; done; psql xpub xpub -c \"SELECT 'public.entities'::regclass\""; then
```

Comment:
I think `/dev/tcp/postgres/5432` has to be substituted with `/dev/tcp/$PGHOST/$PGPORT` here (we wait for the right database to be listening).

By this time they should be available if the `psql` command is going to work

Reply:
(there are two `until echo > /dev/tcp/postgres/5432;` but we may be able to remove the second one since the first has waited already?)

Reply:
I'm not sure I understand this `/dev/tcp/$PROCESS/$PORT` idiom... is this something that will exist for remote databases?

Reply:
I assume you're suggesting this since we've parameterised the host and port using `$PGHOST` and `$PGPORT`?

Reply:
Personally I prefer the use of `netcat`

Reply:
Yes:
```
elife@staging--xpub:/srv/elife-xpub$ docker-compose exec postgres /bin/bash
root@e169bc5bd6a7:/# echo > /dev/tcp/elife-xpub-staging.cxyopn44uqbl.us-east-1.rds.amazonaws.com/5432
root@e169bc5bd6a7:/# echo $?
0
root@e169bc5bd6a7:/# echo > /dev/tcp/localhost/5432
root@e169bc5bd6a7:/# echo $?
0
root@e169bc5bd6a7:/# echo > /dev/tcp/localhost/1234
bash: connect: Cannot assign requested address
bash: /dev/tcp/localhost/1234: Cannot assign requested address
root@e169bc5bd6a7:/# echo $?
1
```
It is not a real file (shares the name), not even a block device file like `/dev/xvda`; it gets intercepted by `bash` and turned into an opened connection to that host and port.

Reply:
> I prefer the use of netcat

It's not as available in containers, whereas `bash` is pretty much always there

Reply:
> since we've parameterised the host and port using $PGHOST and $PGPORT?

yes